### PR TITLE
Attempt to update to use non deprecate file/popen

### LINKED ISSUE #18: file/popen is deprecated
Looks like a small update is warranted

```
/usr/local/lib/janet/mendoza/init.janet:188:15: compile warning (normal): file/popen is deprecated
/usr/local/lib/janet/mendoza/init.janet:199:13: compile warning (normal): file/popen is deprecated
```

This comes up when running `mdz serve`

### PR DESCRIPTION
**Pull request by adam-stokes**
Still learning janet and functional in general, but this is my attempt to fix the deprecated usage of file/popen for the watch command

Fixes #18

### CODE REVIEW

**Comment by pepe** [comment]
I have fetched your branch, and I cannot run the watch. I will check what's up later today.

File: mendoza/init.janet [hunk 1]
```diff
@@ -185,23 +185,19 @@
     # watch runs on linux and mac at least
     # detection is not a solution for windows
     # fswatch technically can run on windows however
-    (def proc (file/popen (string "which " command)))
-    (if (not proc) false
-      (let [result (:read proc :line)]
-        (file/close proc)
-        (and result (not (string/has-suffix? "not found\n" result))))))
+    (def proc (os/spawn ["which" command] :p {:out :pipe}))
+    (= (:wait proc) 0))
```
Comment (adam-stokes) [inline mendoza/init.janet]:
```suggestion
    (zero? (:wait proc))
```

File: mendoza/init.janet [hunk 2]
```diff
@@ -185,23 +185,19 @@
...
     (cond
-      (command-possible "inotifywait") (string "inotifywait -m -r " watched-dirs "-e modify")
-      (command-possible "fswatch") (string "fswatch -r --one-per-batch " watched-dirs)
+      (command-possible "inotifywait") ["inotifywait" "-m" "-r" ;watched-dirs "-e" "modify"]
+      (command-possible "fswatch") ["fswatch" "-r" "--one-per-batch" ;watched-dirs]
       (error "inotifywait, fswatch not available, cannot perform watch.")))
-  (def proc (file/popen cmd :r))
-  (if (not proc) (error (string "could not run " (describe cmd))))
+  (def proc (os/spawn cmd :p {:out :pipe}))
+  (if (not (= (:wait proc) 0)) (error (string "could not run " (describe cmd))))
```
Comment (adam-stokes) [inline mendoza/init.janet]:
```suggestion
  (if (not (zero? (:wait proc))) (error (string "could not run " (describe cmd))))
```

File: mendoza/init.janet (hunk 1 above)
Comment (pepe) [reply mendoza/init.janet]: Yep, I have seen @sogaiu suggesting it, so I would go with them :-)

**Pull request closed without merging**